packages/modeling/src/operations/extrusions/earcut/linkedListSort.js

```diff
@@ -0,0 +1,54 @@
+
+// Simon Tatham's linked list merge sort algorithm
+// http://www.chiark.greenend.org.uk/~sgtatham/algorithms/listsort.html
+const sortLinked = (list) => {
+  let i, p, q, e, numMerges
+  let inSize = 1
+
+  do {
+    p = list
+    list = null
+    let tail = null
+    numMerges = 0
+
+    while (p) {
+      numMerges++
+      q = p
+      let pSize = 0
+      for (i = 0; i < inSize; i++) {
+        pSize++
+        q = q.nextZ
+        if (!q) break
+      }
+
+      let qSize = inSize
+
+      while (pSize > 0 || (qSize > 0 && q)) {
+        if (pSize !== 0 && (qSize === 0 || !q || p.z <= q.z)) {
```

Comment:
Is this the comparison of elements? Is there anyway to break the comparison out into a function?

Reply:
done, add a sorting function parameter